src/lightning_habana/pytorch/datamodule/dataloaders/resnet_media_pipe.py

```diff
@@ -0,0 +1,260 @@
+# Copyright (c) 2023 Habana Labs, Ltd. an Intel Company
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from typing import Any, Tuple
+
+import numpy as np
+import torch.utils.data
+
+import lightning_habana.pytorch.datamodule.utils as utils
+from lightning_habana.utils.imports import _HPU_AVAILABLE
+
+TRAIN_RESIZE_DIM = 224
+EVAL_RESIZE_DIM = 256
+CROP_DIM = 224
+
+DECODER_SCALE_MIN = 0.08
+DECODER_SCALE_MAX = 1.0
+DECODER_RATIO_MIN = 0.75
+DECODER_RATIO_MAX = 1.3333333333333333
+
+USE_HORIZONTAL_FLIP = 1
+FLIP_PROBABILITY = 0.5
+
+RGB_MEAN_VALUES = [0.485, 0.456, 0.406]
+RGB_STD_VALUES = [0.229, 0.224, 0.225]
+RGB_MULTIPLIER = 255
+
+EVAL_CROP_X = 0.5
+EVAL_CROP_Y = 0.5
+
+if _HPU_AVAILABLE:
+    try:
+        from habana_frameworks.mediapipe import fn
+        from habana_frameworks.mediapipe.media_types import decoderStage, dtype, ftype, imgtype, randomCropType
+        from habana_frameworks.mediapipe.mediapipe import MediaPipe
+        from habana_frameworks.mediapipe.operators.cpu_nodes.cpu_nodes import media_function
+        from habana_frameworks.mediapipe.plugins.iterator_pytorch import HPUResnetPytorchIterator
+    except ImportError:
+        raise ModuleNotFoundError("habana_dataloader package is not installed.")
+
+    class ResnetMediaPipe(MediaPipe):
+        """resnet classifier with media pipe.
+
+        :params is_training: True if ResnetMediaPipe handles training data, False in case of evaluation.
+        :params root: path from which to load the images.
+        :params batch_size: mediapipe output batch size.
+        :params shuffle: whether images have to be shuffled.
+        :params drop_last: whether to drop the last incomplete batch or round up.
+        :params queue_depth: Number of preloaded image batches for every slice in mediapipe. <1/2/3>
+        :params num_instances: number of devices.
+        :params instance_id: instance id of current device.
+        :params device: media device to run mediapipe on. <hpu/hpu:0>
```

Comment:
```suggestion
        Args:
            is_training: True if ResnetMediaPipe handles training data, False in case of evaluation.
            root: path from which to load the images.
            batch_size: mediapipe output batch size.
            shuffle: whether images have to be shuffled.
            drop_last: whether to drop the last incomplete batch or round up.
            queue_depth: Number of preloaded image batches for every slice in mediapipe. <1/2/3>
            num_instances: number of devices.
            instance_id: instance id of current device.
            device: media device to run mediapipe on. <hpu/hpu:0>
```